### ISSUE TITLE
How do you evaluate a model for semantic search in huge corpora?

### ISSUE THREAD

**Issue by ErfolgreichCharismatisch** [UKPLab/sentence-transformers#1464]
To preface, I am not a computer scientist. But I read a thing or two.

What I came across was the term "ground truth", which seems to be a reference to determine the best outcome of a model.

Like a picture that you try to copy with water colors and after that you compare your painting to the picture and see based on the differences how good you are.

Makes sense.

But what is ground truth here?

After all, the model determines logical connections between snippets. Is snippet a similar to b, are they dealing with the same topic?

This might be feasible if you offer an array of three sentences, 5 words long each, and one query of 5 words.

But if you have huge corpora of millions, how do you even know the best snippet to match your query?

In other words, in huge corpora, what is your ground truth? Maybe the model finds a "better ground truth"?

**Comment by nreimers** [comment]
This is a challenge. You will not find the perfect snippet.

Typically you use some different systems, retrieve hits for the query, and then annotate them for relevance. But it is very common that you miss good or better hits.
